primitives/src/merkle_tree/field_based_mht/smt/big_lazy_merkle_tree.rs

```diff
@@ -0,0 +1,987 @@
+                return Some(retrieved_elem);
+            },
+            Ok(None) => {
+                return None;
+            },
+            Err(e) => {
+                println!("operational problem encountered: {}", e);
+                return None;
+            },
+        }
+    }
+
+    pub fn remove_from_cache(&self, coord: Coord) -> Option<F>{
+        let coordinates = bincode::serialize(&coord).unwrap();
+        match self.database.get(coordinates.clone()) {
+            Ok(Some(value)) => {
+                let retrieved_elem = F::read(value.as_slice()).unwrap();
+                let res = self.database.delete(coordinates.clone());
+                match res {
+                    Ok(_) => {
+                        return Some(retrieved_elem);
+                    },
+                    Err(e) => {
+                        println!("Could not delete node from cache: {}", e);
+                        return None;
+                    }
+                }
+            },
+            Ok(None) => {
+                return None;
+            },
+            Err(e) => {
+                println!("operational problem encountered: {}", e);
+                return None;
+            },
+        }
+    }
+
+    pub fn insert_to_db(&self, idx: usize, data: F) {
+        let elem = to_bytes!(data).unwrap();
+        let index = bincode::serialize(&idx).unwrap();
+        self.database.put(index, elem).unwrap();
+    }
+
+    pub fn get_from_db(&self, idx: usize) -> Option<F>{
+        let index = bincode::serialize(&idx).unwrap();
+        match self.database.get(index) {
+            Ok(Some(value)) => {
+                let retrieved_elem = F::read(value.as_slice()).unwrap();
+                return Some(retrieved_elem);
+            },
+            Ok(None) => {
+                return None;
+            },
+            Err(e) => {
+                println!("operational problem encountered: {}", e);
+                return None;
+            },
+        }
+    }
+
+    pub fn remove_from_db(&self, idx: usize) -> Option<F>{
+        let index = bincode::serialize(&idx).unwrap();
+        match self.database.get(index.clone()) {
+            Ok(Some(value)) => {
+                let retrieved_elem = F::read(value.as_slice()).unwrap();
+                let res = self.database.delete(index.clone());
+                match res {
+                    Ok(_) => {
+                        return Some(retrieved_elem);
+                    },
+                    Err(e) => {
+                        println!("Could not delete leaf from db: {}", e);
+                        return None;
+                    }
+                }
+
+            },
+            Ok(None) => {
+                return None;
+            },
+            Err(e) => {
+                println!("operational problem encountered: {}", e);
+                return None;
+            },
+        }
+    }
+
+    pub fn check_b_plus_caching_level_down(&mut self, coord: Coord) {
```

Comment:
Can't we use this function also in BigMerkleTree ?

Reply:
I think we can use it. In BigMerkleTree these checks are included inside the function but it could be clearer to put it outside.